src/ast/node.ts

```diff
@@ -40,3 +43,11 @@ export const isFieldNode = (node: SelectionNode): node is FieldNode =>
 export const isInlineFragment = (
   node: SelectionNode
 ): node is InlineFragmentNode => node.kind === Kind.INLINE_FRAGMENT;
+
+export const unwrapType = (
```

Comment:
This probably makes sense to be declared as a generic

```ts
const unwrapType = <T, V = T extends GraphQLWrappingType<infer R> ? R : T>(type: T): V {
 ...
}
```

Reply:
Tried that but `GraphQLWrappingType` doesn't have a generic. But given that we know that we'll get (our own) `GraphQLFlatType` it wouldn't make it stronger :+1: